ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -107,6 +107,8 @@ public class VcrReplicationManager extends ReplicationEngine {
   private DistributedLock vcrUpdateDistributedLock = null;
   private final List<ReplicaThread> threadPool;
   private final AtomicInteger threadIndex;
+  private final HashMap<Long, ReplicaThread> partitionIdThreadMap;
+  private final ReadWriteLock partitionIdThreadMapLock = new ReentrantReadWriteLock();
```

Comment:
We probably don't need this new lock , as we have a rwLock defined in ReplicationEngine.

Reply:
yeah, agree, more locks creates a potential for deadlocks

Reply:
removed the lock and the thread-map, tracking replicaThread in partitionInfo.